src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/TextEditResponseRewriter.cs

```diff
@@ -41,9 +41,10 @@ internal class TextEditResponseRewriter : DelegatedCompletionResponseRewriter
             }
             else
             {
-                // TO-DO: Handle InsertReplaceEdit type
-                // https://github.com/dotnet/razor/issues/8829
-                Debug.Fail("Unsupported edit type.");
+                var insertReplaceRange = editRange.Second.AssumeNotNull();
```

Comment:
The `.AssumeNotNull()` here is confusing me. I _think_ the `.Second` property will throw if the type isn't correct, and the check on line 36 I would have thought would catch the null. What is the `SumType` here exactly?

Perhaps its better to do `else if (editRange.TryGetSecond(out var insertReplaceRange)` and leave an else for catching unsupported types?

Reply:
honestly I didn't know `SumType` threw on property edit. Likely `AssumeNotNull` is correctly not needed